### ISSUE TITLE
False negatives for ts code in `prefer-result-array-groups`

### ISSUE THREAD

**Issue by ota-meshi** [ota-meshi/eslint-plugin-regexp#355]
The `prefer-result-array-groups` rule does not report some problem. I think it's probably because it doesn't handle the TS AST well.

```ts
const parts = /(?<foo>foo)/u.exec(str)!
const foo = parts[1] // False negative
```

---

Also, it seems that we may not be able to autofix the problem.

```ts
/^Readonly(?<typeName>.*)/.exec(name)?.[1]
```

Probably the array type will be `RegExpExecArray | null` and will stop autofix.
